security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -81,14 +82,14 @@ func NewSecretCache(cl CAClient, secretTTL, rotationInterval, evictionDuration t
 // GetSecret gets secret from cache, this function is called by SDS.FetchSecret,
 // Since credential passing from client may change, regenerate secret every time
 // instread of reading from cache.
-func (sc *SecretCache) GetSecret(proxyID, token string) (*sds.SecretItem, error) {
-	ns, err := sc.generateSecret(token, time.Now())
+func (sc *SecretCache) GetSecret(proxy model.Proxy, token string) (*sds.SecretItem, error) {
```

Comment:
I don't think you need to change this to `model.Proxy`, given we don't need IP address.

proxyID works... but since coneptually this is a map from identity URI to the certificate/key, suggest to use spiffeID or identity as the key

Reply:
From [BuildSubjectAltNameExtension implementation](https://github.com/istio/istio/blob/97882cede142f5ac246a2e7cff4d6ff6bf631389/security/pkg/pki/util/san.go#L79), either service account or IP address could be used in csr template;  any reason you prefer service account ? and how to get the service account in sds ?

Reply:
One place where the service account name will be available is in the Pod annotation. Do you have access to Pod annotations here?

Reply:
I think so (https://github.com/istio/istio/blob/5c20da865125980921d2652e31fee47f19d2af73/pilot/pkg/serviceregistry/kube/conversion.go#L122).

After discussing offline and prototype, we decides to use service account to move forward, thanks !